repository.py

```diff
@@ -331,6 +334,25 @@ def stage_cli(
             dry_run=dry_run,
         )
 
+def gen_token_cli(
+    slurm_charms: [str],
+    **kwargs,
+):
+    """Generate Charmhub token to publish charms."""
+    args = [
+        "charmcraft",
+        "login",
+        "--export=slurm-charms.auth"
+    ] + [f"--charm={charm}" for charm in slurm_charms] + [
+        "--permission=package-manage-metadata",
+        "--permission=package-manage-releases",
+        "--permission=package-manage-revisions",
+        "--permission=package-view-metadata",
+        "--permission=package-view-releases",
+        "--permission=package-view-revisions",
+        "--ttl=7776000"
```

Comment:
How do we feel about adding the comment 90 days so we have a better idea when the token is going to expire :sweat_smile: 

```suggestion
        "--ttl=7776000"  # 90 days
```